spec/factories/vacancies.rb

```diff
@@ -44,7 +48,7 @@
     expires_at { 6.months.from_now.change(hour: 9, minute: 0, second: 0) }
     hired_status { nil }
     include_additional_documents { false }
-    sequence :job_title, job_titles.cycle
+    job_title { Rails.env.production? ? factory_sample(job_titles) : generate(:job_title) }
```

Comment:
Only calling the sequence in non-prod envs.
So the current production  job application example through the factory bot will keep looking "pretty", while testing will generate sequential job titles to avoid ambiguous matching errors.